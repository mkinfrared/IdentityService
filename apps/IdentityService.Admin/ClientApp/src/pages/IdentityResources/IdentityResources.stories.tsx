import { Meta, StoryObj } from "@storybook/react";

import IdentityResources from "./IdentityResources";
import css from "./Story.module.scss";

type Story = StoryObj<typeof meta>;

const meta = {
  title: "UI/IdentityResources",
  component: IdentityResources,
  tags: ["autodocs"],
} satisfies Meta<typeof IdentityResources>;

const Default: Story = {
  args: {},
  render: (args) => (
    <div className={css.Story}>
      <IdentityResources {...args} />
    </div>
  ),
};

export { Default };

export default meta;
