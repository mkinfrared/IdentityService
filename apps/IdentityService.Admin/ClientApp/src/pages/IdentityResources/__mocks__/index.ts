export { default } from "./IdentityResources.mock";
