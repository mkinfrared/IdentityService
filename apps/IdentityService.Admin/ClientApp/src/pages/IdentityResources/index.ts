export { default } from "./IdentityResources";

export * from "./IdentityResources.type";
