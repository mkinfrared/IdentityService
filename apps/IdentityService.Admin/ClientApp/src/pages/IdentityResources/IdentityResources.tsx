import css from "./IdentityResources.module.scss";
import { IdentityResourcesProps } from "./IdentityResources.type";

const IdentityResources = ({ className }: IdentityResourcesProps) => {
  const classNames = [css.IdentityResources, className]
    .filter(Boolean)
    .join(" ");

  return (
    <div className={classNames} data-testid="IdentityResources">
      IdentityResources works!!
    </div>
  );
};

export default IdentityResources;
