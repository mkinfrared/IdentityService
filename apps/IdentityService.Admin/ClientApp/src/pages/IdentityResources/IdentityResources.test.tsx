import { render } from "@testing-library/react";

import IdentityResources from "./IdentityResources";

describe("<IdentityResources />", () => {
  const Component = <IdentityResources />;

  it("should be defined", () => {
    expect(IdentityResources).toBeDefined();
  });

  it("should match the snapshot", () => {
    const { container } = render(Component);

    expect(container).toMatchSnapshot();
  });

  it("should contain a data test id", () => {
    const { getByTestId } = render(Component);
    const element = getByTestId("IdentityResources");

    expect(element).toBeDefined();
  });
});
