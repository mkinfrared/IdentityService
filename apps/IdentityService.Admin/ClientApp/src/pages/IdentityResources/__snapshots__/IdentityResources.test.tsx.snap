// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IdentityResources /> should match the snapshot 1`] = `
<div>
  <div
    class="IdentityResources"
    data-testid="IdentityResources"
  >
    IdentityResources works!!
  </div>
</div>
`;
